zerver/actions/message_send.py

```diff
@@ -623,6 +625,12 @@ def build_message_send_dict(
         possible_stream_wildcard_mention=mention_data.message_has_stream_wildcards(),
     )
 
+    default_code_block_language = ""
+    if stream is not None:
+        # We need to use the default code block language of the stream
+        # for rendering the message.
+        default_code_block_language = stream.default_code_block_language
```

Comment:
Would be good to add a test for this checking which language is used to render code blocks when sending messages to streams with different value set and when sending DMs.